### PR TITLE
Add typedspark_schema property to a dataset

### LINKED ISSUE #138: Expose the schema of a dataset to allow for functions to work on any dataset.

### PR DESCRIPTION
**Pull request by hahamark1**
Fixes #138 

This PR adds a property on a dataset to expose the provided schema.

### CODE REVIEW

File: typedspark/_core/dataset.py [hunk 1]
```diff
@@ -86,6 +86,11 @@ def _add_schema_metadata(self) -> None:
         for field in self._schema_annotations.get_structtype().fields:
             self.schema[field.name].metadata = field.metadata
 
+    @property
+    def typedspark_schema(self) -> Schema:
```
Comment (nanne-aben) [inline typedspark/_core/dataset.py:90]: Kunnen we het return-type hier `T` maken? From `DataSet[T]`.

**Review by nanne-aben — COMMENTED** [review]
Nice! Left a small comment.

File: typedspark/_core/dataset.py (hunk 1 above)
Comment (hahamark1) [reply typedspark/_core/dataset.py]: Yes!

Comment (hahamark1) [reply typedspark/_core/dataset.py]: The typechecker doesnt like this..

Comment (nanne-aben) [reply typedspark/_core/dataset.py]: Just leave a `#type: ignore` at the return statement

**Review by nanne-aben — APPROVED** [review]

File: typedspark/_core/dataset.py (hunk 1 above)
Comment (hahamark1) [reply typedspark/_core/dataset.py]: Taking a second look, shouldnt it be `Type[T]`?

Comment (nanne-aben) [reply typedspark/_core/dataset.py]: Ah, yeah, you're right!

Comment (hahamark1) [reply typedspark/_core/dataset.py]: So its wrong in other places aswell?

**Pull request merged**